src/shared/exec-util.c

```diff
@@ -445,9 +446,9 @@ ExecCommandFlags exec_command_flags_from_string(const char *s) {
 }
 
 int fexecve_or_execve(int executable_fd, const char *executable, char *const argv[], char *const envp[]) {
-        fexecve(executable_fd, argv, envp);
-        if (errno == ENOENT)
-                /* A script? Let's fall back to execve().
+        execveat(executable_fd, "", argv, envp, AT_EMPTY_PATH);
+        if (IN_SET(errno, ENOSYS, ENOENT))
```

Comment:
probably should handle EPERM too, to deal with seccomp filtering container managers.

```c
if (ERRNO_IS_NOT_SUPPORTED(errno) || ERRNO_IS_PRIVILEGED(errno) || errno == ENOENT)
```

maybe?

Reply:
It's
```C
if (IN_SET(errno, ENOSYS, ENOENT) || ERRNO_IS_PRIVILEGE(errno))
```
now, so that seems to be handled. (Not sure when this comment was made. Github shows it as recent, but outdated.)